Prox/Prox/PlaceCarousel/PlaceCarouselCollectionViewCell.swift

```diff
@@ -0,0 +1,97 @@
+        let view = PlaceReviewView()
+        view.color = .red
+        view.translatesAutoresizingMaskIntoConstraints = false
+        return view
+    }()
+
+    lazy var tripAdvisorReview: PlaceReviewView = {
+        let view = PlaceReviewView()
+        view.color = .green
+        view.translatesAutoresizingMaskIntoConstraints = false
+        return view
+    }()
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        
+        setupViews()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private func setupViews() {
+        contentView.addSubview(roundedBackgroundView)
+
+        var constraints = [roundedBackgroundView.topAnchor.constraint(equalTo: contentView.topAnchor),
+                           roundedBackgroundView.leadingAnchor.constraint(equalTo: contentView.leadingAnchor),
+                           roundedBackgroundView.trailingAnchor.constraint(equalTo: contentView.trailingAnchor),
+                           roundedBackgroundView.bottomAnchor.constraint(lessThanOrEqualTo: contentView.bottomAnchor)]
+
+        roundedBackgroundView.addSubview(placeImage)
+        constraints.append(contentsOf: [placeImage.topAnchor.constraint(equalTo: roundedBackgroundView.topAnchor),
+                                        placeImage.leadingAnchor.constraint(equalTo: roundedBackgroundView.leadingAnchor),
+                                        placeImage.trailingAnchor.constraint(equalTo: roundedBackgroundView.trailingAnchor),
+                                        placeImage.heightAnchor.constraint(equalToConstant: 201)])
+
+        roundedBackgroundView.addSubview(yelpReview)
+        constraints.append(contentsOf: [yelpReview.topAnchor.constraint(equalTo: placeImage.bottomAnchor, constant: 13),
+                                        yelpReview.leadingAnchor.constraint(equalTo: roundedBackgroundView.leadingAnchor),
+                                        yelpReview.heightAnchor.constraint(equalToConstant: 50),
+                                        yelpReview.widthAnchor.constraint(equalToConstant: 100)])
+
+        roundedBackgroundView.addSubview(tripAdvisorReview)
+        constraints.append(contentsOf: [tripAdvisorReview.topAnchor.constraint(equalTo: placeImage.bottomAnchor, constant: 13),
+                                        tripAdvisorReview.trailingAnchor.constraint(equalTo: roundedBackgroundView.trailingAnchor),
+                                        tripAdvisorReview.heightAnchor.constraint(equalToConstant: 50),
+                                        tripAdvisorReview.widthAnchor.constraint(equalToConstant: 100)])
+
+        NSLayoutConstraint.activate(constraints)
+    }
+
+    override func layoutSubviews() {
+        super.layoutSubviews()
+
+        NSLayoutConstraint.activate([roundedBackgroundView.heightAnchor.constraint(equalToConstant: contentView.frame.size.height),
```

Comment:
yes, in setup views, the frame is set to zero making querying it for actual sizes impossible. We have to wait until the view are actually laid out before we get actual resolved view sizes. In `setupViews` we can sort out relative layout, but any constant layout that requires reference to the frame will have to be done in `layoutSubviews`.